Resources/public/js/ezrecommendation.js

```diff
@@ -0,0 +1,77 @@
+// register HandleBars helpers when library is loaded
+
+document.addEventListener('DOMContentLoaded', function() {
+    Handlebars.registerHelper('each', function(context, options) {
+        var result = '';
+        for(var i = 0, j = context.length; i < j; i++) {
+            result = result + options.fn(context[i]);
+        }
+        return result;
+    });
+
+    Handlebars.registerHelper('formatDate', function(timestamp) {
+        var fixedTimestamp = (timestamp + '000') * 1;
+        var t = new Date(fixedTimestamp);
+        return t.toLocaleDateString();
+    });
+
+    Handlebars.registerHelper('formatTime', function(timestamp) {
+        var fixedTimestamp = (timestamp + '000') * 1;
+        var t = new Date(fixedTimestamp);
+        return t.toLocaleTimeString();
+    });
+});
+
+// EzRecoRestClient class
+
+var EzRecoRestClient = function (config) {
+    this.config = config;
+};
+
+EzRecoRestClient.prototype.fetchRecommendations = function (responseCallback, targetId, templateId) {
+    var xmlhttp = new XMLHttpRequest();
+    var jsonResponse;
+
+    xmlhttp.onreadystatechange = function () {
+        if (xmlhttp.readyState === XMLHttpRequest.DONE) {
+            if (xmlhttp.status == 200) {
+                jsonResponse = JSON.parse(xmlhttp.response);
+            } else {
+                jsonResponse = new Array();
+                jsonResponse['status'] = 'fail';
```

Comment:
so jsonReponse is not an array, those 2 lines should be written: `jsonResponse = {'status': 'fail'};`